tests/generators/test_generator.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+"""Tests for the :mod:`aiida_common_workflows.generators.generator` module."""
+import pytest
+from aiida import orm
+from aiida.plugins import WorkflowFactory
+
+
+def test_instan_inputgen(test_inputgen):
+    """
+    Test the instanciation of a subclass ot `InputGenerator`
+    """
+
+    clas = test_inputgen()
+    with pytest.raises(RuntimeError):
+        clas()
+
+    clas(process_class=WorkflowFactory('common_workflows.relax.siesta'))
+
+
+def test_serializer(test_inputgen, generate_structure):
+    """
+    Test the serializer of the arguments of `get_builder`
+    """
+
+    clas = test_inputgen(inputs_dict={'structure': orm.StructureData})
+
+    class_ins = clas(process_class=WorkflowFactory('common_workflows.relax.siesta'))
+
+    in_struct = generate_structure(symbols=('Si',))
+    buil, args = class_ins.get_builder(structure=in_struct)  # pylint: disable=unused-variable
+    # in_struct is an unstored node, therefore it is deepcopied and
+    # for this reason the uuid of the initial structure and the one
+    # inside `get_builder` are different
+    assert args['structure'].uuid != in_struct.uuid
+
+    in_struct = generate_structure(symbols=('Si',)).store()
+    buil, args = class_ins.get_builder(structure=in_struct)  # pylint: disable=unused-variable
+    # in_struct is now stored, therefore it is maintained
+    # within `get_builder`
+    assert args['structure'].uuid == in_struct.uuid
```

Comment:
Removed the fact that the `get_builder` also returns the arguments as that is changing the signature and is not actually necessary. We want to check that the `kwargs` that are passed to it are not modified, so we can check that directly.
```suggestion
    cls = generate_input_generator_cls(inputs_dict={'structure': orm.StructureData})
    generator = cls(process_class=WorkflowFactory('common_workflows.relax.siesta'))

    structure = generate_structure(symbols=('Si',))
    kwargs = {'structure': structure}
    generator.get_builder(**kwargs)
    # structure is an unstored node, therefore it is deepcopied and so the UUID should change.
    assert kwargs['structure'].uuid != structure.uuid

    structure = generate_structure(symbols=('Si',)).store()
    kwargs = {'structure': structure}
    generator.get_builder(**kwargs)
    # The structure is now stored and so should be the same.
    assert kwargs['structure'].uuid == structure.uuid
```

Reply:
I am not totally sure about this @sphuber. I wanted to check that the kwargs are deepcopied inside the `get_builder` if they are not stored nodes. Are you instead proposing to change one of the  kwargs (for instance structure) inside the `get_builder` and see that the original did not change??